database/buildSchema/35_file.sql

```diff
@@ -5,7 +5,7 @@ CREATE TABLE public.file (
     original_filename varchar NOT NULL,
     user_id integer REFERENCES public.user (id) ON DELETE CASCADE,
     template_id integer REFERENCES public.template (id) ON DELETE CASCADE,
-    application_serial varchar REFERENCES public.application (serial) ON DELETE CASCADE,
+    application_serial varchar REFERENCES public.application (serial) ON DELETE CASCADE ON UPDATE CASCADE,
```

Comment:
We should have done this a long time ago. Without it, if you change the application serial and there's a file that points to it (via serial), it'll fail with an error. Now it'll update along with it :)

Reply:
Cool! I noticed something similar for when we change the Template.code and it doesn't reflect on template_element and other places that use the combination of Template.code and Template.version as reference. It doesn't get updated. Maybe we can also change that now.

Reply:
That's a separate issue though, as the "template_code" and "version" fields on template_element are GENERATED columns, not foreign key references. And STORED Generated columns get generated when first instantiated, not again.

I'm not sure how often this is a problem. If we're querying through GraphQL it's fine, as we'll drill down through the foreign keys from "template".

Do you know where the "template_code" field in "template_element" is being used?

If it's important, we may have to create a trigger to update it.